prov/util/src/util_av.c

```diff
@@ -1465,17 +1469,13 @@ void ofi_cmap_process_reject(struct util_cmap *cmap,
 	cmap->acquire(&cmap->lock);
 	switch (handle->state) {
 	case CMAP_CONNREQ_RECV:
+	case CMAP_CONNREQ_SENT:
 	case CMAP_CONNECTED:
 	case CMAP_CONNECTED_NOTIFY:
 		/* Handle is being re-used for incoming connection request */
 		FI_DBG(cmap->av->prov, FI_LOG_EP_CTRL,
 			"Connection handle is being re-used. Ignoring reject\n");
 		break;
-	case CMAP_CONNREQ_SENT:
-		FI_DBG(cmap->av->prov, FI_LOG_EP_CTRL,
-			"Deleting connection handle\n");
-		util_cmap_del_handle(handle);
-		break;
```

Comment:
You never know if the reject received is because of simultaneous connections or a genuine reject. In case of a client-server test, the client may hang because it's stuck at CMAP_CONNREQ_SENT state. We could potentially use CM data in reject request to avoid ambiguity.